cl-postgres/interpret.lisp

```diff
@@ -92,6 +92,47 @@ interpreted as an array of the given type."
 (define-interpreter 1042 "bpchar" string)
 (define-interpreter 1043 "varchar" string)
 
+;; HACK ALERT: we need to squirrel away the current socket when
+;; processing queries. Define this here as interpret.lisp is compiled
+;; before protocol.lisp
+(defvar *current-socket*)
```

Comment:
I think that if you need to use hacks like this, you're just at the wrong abstraction level. I'd prefer if you'd use `set-sql-reader` directly instead of badly fitting this onto `define-interpreter`.